### PR TITLE
feat: support `2+2` and `/status/buildinfo`

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
Closes https://github.com/GreptimeTeam/greptimedb/issues/3496

## What's changed and what's your intention?

When adding GreptimeDB to Grafana (v9.0) datasource using Prometheus extension, it will send two queries:
```
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on docker0, link-type EN10MB (Ethernet), snapshot length 262144 bytes
15:55:50.899080 IP 172.17.0.3.54034 > 172.17.0.1.4000: Flags [P.], seq 3048369435:3048369718, ack 3792103051, win 251, options [nop,nop,TS val 2511689527 ecr 2125370641], length 283
E..O..@.@.................m.........Yh.....
..W7~...POST /v1/prometheus/api/v1/query HTTP/1.1
Host: 172.17.0.1:4000
User-Agent: Grafana/10.2.3
Content-Length: 18
Content-Type: application/x-www-form-urlencoded
X-Datasource-Uid: c91e98f4-01ea-4ffd-890a-eef523983c78
X-Grafana-Org-Id: 1
Accept-Encoding: gzip

query=1%2B1&time=4
15:55:50.901986 IP 172.17.0.3.54034 > 172.17.0.1.4000: Flags [P.], seq 283:489, ack 229, win 250, options [nop,nop,TS val 2511689530 ecr 2125370644], length 206
E.....@.@.................n6...o....Y......
..W:~...GET /v1/prometheus/api/v1/status/buildinfo HTTP/1.1
Host: 172.17.0.1:4000
User-Agent: Grafana/10.2.3
X-Datasource-Uid: c91e98f4-01ea-4ffd-890a-eef523983c78
X-Grafana-Org-Id: 1
Accept-Encoding: gzip
```

This patch supports these two requests. Now GreptimeDB can be add to Grafana data source via Prometheus extension without error:

image

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
Thank you!

File: src/servers/src/http/prometheus_resp.rs [hunk 1]
```diff
@@ -256,24 +258,35 @@ impl PrometheusJsonResponse {
...
-            })
-            .collect::<Result<Vec<_>>>()?;
+                PromQueryResult::Matrix(ref mut v) => {
+                    v.push(PromSeriesMatrix { metric, values });
+                }
+                PromQueryResult::Scalar(ref mut v) => {
+                    *v = values.pop();
+                }
+                PromQueryResult::String(ref mut _v) => {
+                    // Not supported yet
```
Comment (killme2008) [inline src/servers/src/http/prometheus_resp.rs:286]: Adds a `TODO`?

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/servers/src/http/prometheus_resp.rs (hunk 1 above)
Comment (waynexia) [inline src/servers/src/http/prometheus_resp.rs:286]:
```suggestion
                    // TODO(ruihang): Not supported yet
```

**Pull request merged**